### PR TITLE
PyTorch example

### PR DESCRIPTION
**Pull request by martinengelcke**
#4 #6

### CODE REVIEW

File: forge/examples/torch/train.py [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+
+# Load data
+train_loader = fet.load(config.data_config, config)
+# Load model
+model = fet.load(config.model_config, config)
+
+# Print flags
+fet.print_flags()
+# Print model info
+print(model)
```
Comment (akosiorek) [inline forge/examples/torch/train.py:75]: Oh that looks fun. Is it a pytorch feature?

File: forge/examples/torch/train.py [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+
+# Setup optimizer
+optimizer = optim.RMSprop(model.parameters(),
+                          lr=config.learning_rate,
+                          momentum=0.9)
+
+# Try to restore model and optimizer from checkpoint
+if resume_checkpoint is not None:
+    print("Restoring checkpoint from '{}'".format(resume_checkpoint))
```
Comment (akosiorek) [inline forge/examples/torch/train.py:84]: would be nice to add a utility function to do this

File: forge/examples/torch/train.py [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+    start_epoch = 1
+print("Starting training at iter = {}".format(start_epoch))
+
+# Training
+for epoch in range(start_epoch, config.train_epochs + 1):
+    model.train()
+    for batch_idx, (data, target) in enumerate(train_loader):
+        optimizer.zero_grad()
+        output = model(data.view(-1, 784))
+        loss = F.cross_entropy(output, target)
```
Comment (akosiorek) [inline forge/examples/torch/train.py:102]: What's `F`?

**Comment by akosiorek** [comment]
I've left minor in-line comments. LGTM!

**Review by akosiorek — APPROVED** [review]

File: forge/examples/torch/train.py (hunk 1 above)
Comment (martinengelcke) [reply forge/examples/torch/train.py:75]: It is :) Comparable to model.summary() in Keras

File: forge/examples/torch/train.py (hunk 2 above)
Comment (martinengelcke) [reply forge/examples/torch/train.py:84]: Agreed. Will add things like this over time as needed

File: forge/examples/torch/train.py (hunk 3 above)
Comment (martinengelcke) [reply forge/examples/torch/train.py:102]: This is from "import torch.nn.functional as F"

torch.nn defines (neural network) modules as classes and torch.nn.functional is the functional view. I typically use torch.nn for modules that have parameters (e.g. nn.Linear, nn.Conv2d) and torch.nn.functional for parameter-free "layers" (e.g. F.relu, F.cross_entropy)

**Comment by martinengelcke** [comment]
Great, thanks. Will merge :)

File: forge/examples/torch/train.py (hunk 3 above)
Comment (akosiorek) [reply forge/examples/torch/train.py:102]: I'd suggest using at least 3-letter names, as per python style guide.

Comment (martinengelcke) [reply forge/examples/torch/train.py:102]: This is quite a standard naming convention in the PyTorch community

**Pull request merged**